MHVLib/HTTP/MHVHttpService.m

```diff
@@ -0,0 +1,276 @@
+          {
+              //Update offset and call this method again to send the next chunk
+              [self uploadBlobSource:blobSource
+                               toUrl:url
+                         chunkOffset:chunkOffset + thisChunkSize
+                           chunkSize:chunkSize
+                          completion:completion];
+              return;
+          }
+          else
+          {
+              if (error)
+              {
+                  [Logger write:[NSString stringWithFormat:@"Blob upload error: %@", error.localizedDescription]];
+                  
+                  completion(nil, error);
+              }
+              else
+              {
+                  [Logger write:[NSString stringWithFormat:@"Blob upload complete"]];
+                  
+                  completion([self responseFromData:data urlResponse:response], nil);
+              }
+          }
+      }] resume];
+}
+
+#pragma mark - Helpers
+
+- (NSURLRequest *)requestWithUrl:(NSURL *)url
+                            data:(NSString *)dataString
+{
+    NSMutableURLRequest *request = [[NSMutableURLRequest alloc] initWithURL:url];
+    
+    [request setValue:@"gzip" forHTTPHeaderField:@"Accept-Encoding"];
+    
+    if (dataString)
+    {
+        NSData *data = [dataString dataUsingEncoding:NSUTF8StringEncoding];
+        
+        request.HTTPMethod = @"POST";
+        request.HTTPBody = data;
+        
+        [request setValue:[NSString stringWithFormat:@"%lu", (unsigned long)data.length] forHTTPHeaderField:@"Content-Length"];
+    }
+    
+    return request;
+}
+
+- (NSURLRequest *)requestWithUrl:(NSURL *)url
+                            data:(NSData *)data
+                     chunkOffset:(NSUInteger)chunkOffset
+                       totalSize:(NSUInteger)totalSize
+{
+    NSMutableURLRequest *request = [[NSMutableURLRequest alloc] initWithURL:url];
+    
+    request.HTTPMethod = @"POST";
+    request.HTTPBody = data;
+
+    [request setValue:@"gzip" forHTTPHeaderField:@"Accept-Encoding"];
+    [request setValue:@"application/octet-stream" forHTTPHeaderField:@"Content-Type"];
+
+    NSString *contentRange = [NSString stringWithFormat:@"bytes %lu-%lu/*", (unsigned long)chunkOffset, (unsigned long)chunkOffset + data.length - 1];
+    [request setValue:contentRange forHTTPHeaderField:@"Content-Range"];
+
+    //Last chunk, add the "complete" header
+    if (chunkOffset + data.length == totalSize)
+    {
+        [request setValue:@"1" forHTTPHeaderField:@"x-hv-blob-complete"];
+    }
+
+    return request;
+}
+
+- (MHVHttpServiceResponse *)responseFromData:(NSData *)data urlResponse:(NSURLResponse *)response
+{
+    NSInteger statusCode = ((NSHTTPURLResponse *)response).statusCode;
+    
+    return [[MHVHttpServiceResponse alloc] initWithResponseData:data
+                                               statusCode:statusCode];
+}
+
+#pragma mark - NSURLSessionDelegate
+
+- (void)     URLSession:(NSURLSession *)session
```

Comment:
The extra space here just looks bad, I know we are trying to line up the colons, but this just doesn't look right.